### PR TITLE
LLM-224 - Enable maven publishing for JitPack

### PR DESCRIPTION
**Pull request by NickSeagull**
This is enough for JitPack to pickup the builds in theory

### CODE REVIEW

File: modules/embeddings-model/build.gradle.kts [hunk 1]
```diff
@@ -1,8 +1,10 @@
+group = "com.theagilemonkeys.llmental.embeddings-model"
```
Comment (javiertoledo) [inline modules/embeddings-model/build.gradle.kts]: Are dashes allowed/idiomatic here? Maybe...

```suggestion
group = "com.theagilemonkeys.llmental.embeddingsModel"
```

File: modules/semantic-search/build.gradle.kts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+group = "com.theagilemonkeys.llmental.semantic-search"
```
Comment (javiertoledo) [inline modules/semantic-search/build.gradle.kts]: Maybe...

```suggestion
group = "com.theagilemonkeys.llmental.semanticSearch"
```

File: modules/vector-store/build.gradle.kts [hunk 3]
```diff
@@ -1,8 +1,10 @@
+group = "com.theagilemonkeys.llmental.vector-store"
```
Comment (javiertoledo) [inline modules/vector-store/build.gradle.kts]: You know...

```suggestion
group = "com.theagilemonkeys.llmental.vectorStore"
```

File: semantic-search-service-demo/build.gradle.kts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+group = "com.theagilemonkeys.llmental.semantic-search-service-demo"
```
Comment (javiertoledo) [inline semantic-search-service-demo/build.gradle.kts]: 🤔...

```suggestion
group = "com.theagilemonkeys.llmental.semanticSearchServiceDemo"
```

**Review by javiertoledo — APPROVED** [review]
LGTM, just left a couple of notes about groups naming, just in case that it matters

**Comment by NickSeagull** [comment]
In fact, its neither for you or for me, Java standard establishes that packages have to be all lowercase letters: https://docs.oracle.com/javase/specs/jls/se6/html/packages.html#7.7

**Comment by javiertoledo** [comment]
And who are we to go against the _Java establishment™_

**Comment by NickSeagull** [comment]
Changing everything right now, I'm taking the opportunity to rename llmentalCore to core and move the demo into an examples folder

**Pull request merged**